app/models/job_run.rb

```diff
@@ -0,0 +1,10 @@
+class JobRun < ApplicationRecord
+  belongs_to :bundle_context
+  #! TODO: should we validate at some point that the file exist on disk
+  # validates :output_location, presence=true
```

Comment:
We'll never have a `presence: true` validator here, because all these objects will be valid when passed to the correct job, without yet having any `output_location`.  

I might check/validate any values by defining a "setter" like:
```ruby
def output_location=(val)
  raise ArgumentError, "Path #{val} is not a readable file" unless ...
  super
end
```
Or just compose the validation w/ a conditional like: 
```
validate :output_location_validation, if: Proc.new { |jobrun| jobrun.output_location.present? }

def output_location_validation
  errors.add(:base, "Path #{output_location} is not a readable file") unless ...
end
```

Reply:
I initially had `presence: true` set-up in the model but after talking it over with John, we just commented out that line. I would like to talk with you more about your suggestions.